packages/@aws-cdk/aws-appsync/lib/private/schema-implementation.ts

```diff
@@ -0,0 +1,326 @@
+import { readFileSync } from 'fs';
+import { IBucket } from '@aws-cdk/aws-s3';
+import { Lazy } from '@aws-cdk/core';
+import { CfnGraphQLSchema } from '../appsync.generated';
+import { GraphqlApi } from '../graphqlapi';
+import { shapeAddition } from '../private/schema-shape-addition';
+import { Schema } from '../schema';
+import { IIntermediateType } from '../schema-base';
+import { ResolvableField } from '../schema-field';
+import { ObjectType } from '../schema-intermediate';
+
+/**
+ * Utility enum for Schema class
+ */
+enum SchemaMode {
+  FILE = 'FILE',
+  CODE = 'CODE',
+  S3 = 'S3',
+};
+
+export class SchemaImpl implements Schema {
+
+  /**
+   * The definition for this schema
+   */
+  public readonly definition: string;
+
+  /**
+    * The underlying CFN GraphQL Schema Resource. This
+    * resource is created on `bind` which allows the Schema
+    * to be created outside the CDK (a feature that works well
+    * with the code-first approach).
+    *
+    * @internal
+    */
+  protected _schema!: CfnGraphQLSchema;
+
+  /**
+    * The SchemaMode utilized by the Schema Implementation
+    *
+    * @internal
+    */
+  protected _mode: SchemaMode;
+
+  public constructor() {
+    this.definition = '';
+    this._mode = SchemaMode.FILE;
+  }
+
+  /**
+    * Called when the GraphQL Api is initialized to allow this object to bind
+    * to the stack.
+    */
+  public bind(api: GraphqlApi): CfnGraphQLSchema {
+    if (!this._schema) {
```

Comment:
I dont want to create two schemas. So the conditional checks if it was created already.

Reply:
Why?

Reply:
I mean.. if there is not point in it, why take up the memory to run it again? And I think the schema should be the same no? Unless there is some use case where you want to make multiple stacks that each have a schema that builds on top of itself?

Reply:
Look at this code:

```ts
const schema = appsync.Schema.fromCode();
new GraphqlApi(this, 'Api1', { schema, });
new GraphqlApi(this, 'Api2', { schema, });
```

Because of  the caching that you do, the second `Api` does not have a Schema at all (its call to `Schema.bind()` returns the existing `CfnGraphQLSchema` with `apiId` filled with the ID of the first API).

I cant imagine this is the correct behavior.